File: netlify/functions/predict/predict.py
import sys
import json
import joblib
import numpy as np
import os

def predict(data):
    try:
        model_path = os.path.join(os.path.dirname(__file__), 'model.pkl')
        model = joblib.load(model_path)
        
        input_data = np.array(data).reshape(1, -1)
        prediction = model.predict(input_data)[0]
        return {"Recommended Job": str(prediction)}
    except Exception as e:
        return {"error": str(e)}

if __name__ == "__main__":
    input_data = json.loads(sys.argv[1])
    result = predict(input_data)
    print(json.dumps(result))

File: netlify/functions/predict/predict.js
const { spawn } = require("child_process");
const path = require("path");

exports.handler = async function (event, context) {
  if (event.httpMethod !== "POST") {
    return { statusCode: 405, body: "Method Not Allowed" };
  }

  try {
    const data = JSON.parse(event.body);

    // Call Python script with data
    const pythonProcess = spawn("python", [
      path.join(__dirname, "predict.py"),
      JSON.stringify(data.data),
    ]);

    return new Promise((resolve, reject) => {
      let result = "";

      pythonProcess.stdout.on("data", (data) => {
        result += data.toString();
      });

      pythonProcess.stderr.on("data", (data) => {
        console.error(`Error: ${data}`);
      });

      pythonProcess.on("close", (code) => {
        resolve({
          statusCode: 200,
          headers: {
            "Content-Type": "application/json",
          },
          body: result,
        });
      });
    });
  } catch (error) {
    return {
      statusCode: 500,
      body: JSON.stringify({ error: "Failed to process request" }),
    };
  }
};
